vm/src/macros.rs

```diff
@@ -57,6 +57,15 @@ impl MacroEnv {
         self.macros.write().unwrap().insert(name, Box::new(mac));
     }
 
+    /// Inserts a `Macro` then returns an instance of Option<Box<Macro>> using self.get
+    pub fn insert_and_get<M>(&self, name: String, mac: M) -> Option<Box<Macro>>
```

Comment:
No need to return an `Option` here since inserting the macro guarantees that it exists.